src/librustc_mir/interpret/place.rs

```diff
@@ -728,9 +729,8 @@ where
                     _ => bug!("write_immediate_to_mplace: invalid Scalar layout: {:#?}",
                             dest.layout)
                 }
-
                 self.memory.get_mut(ptr.alloc_id)?.write_scalar(
-                    tcx, ptr, ptr_align.min(dest.layout.align), scalar, dest.layout.size
+                    tcx, ptr, scalar, dest.layout.size
```

Comment:
Funny enough, here you got rid of `min`. I approve, but why the inconsistency?

Reply:
I was removing it everywhere, I readded it only after not being able to figure out that I needed `restrict_for_offset`